dutch-auction/src/main.sw

```diff
@@ -0,0 +1,123 @@
+contract;
+
+use std::{
+    address::Address,
+    contract_id::ContractId,
+    block::height,
+    chain::auth::msg_sender,
+    assert::assert,
+    revert::revert,
+    storage::StorageMap,
+    token::transfer_to_output,
+    context::{msg_amount, call_frames::msg_asset_id},
+    result::*,
+    identity::Identity,
+};
```

Comment:
I dont think the `transfer_to` would work here, it only transfers the native token of the current contract to the Identity, not the asset that is being bid with.

But other than that, yeah, I'll add Enum Errors and replace assert with require

Reply:
Oh yeah, I remembered that and then I forgot and got mixed up with a WIP that I'm working on which replaces that function and allows you to transfer any asset instead of `contract_id()` and nothing else